lib/common/tests/rules/pcmk__evaluate_op_expression_test.c

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright 2024 the Pacemaker project contributors
+ *
+ * The version control history for this file may have further details.
+ *
+ * This source code is licensed under the GNU General Public License version 2
+ * or later (GPLv2+) WITHOUT ANY WARRANTY.
+ */
+
+#include <crm_internal.h>
+
+#include <stdio.h>
+#include <glib.h>
+
+#include <crm/common/xml.h>
+#include <crm/common/rules_internal.h>
+#include <crm/common/unittest_internal.h>
+#include "crmcommon_private.h"
+
+/*
+ * Shared data
+ */
+
+static pcmk_rule_input_t rule_input = {
+    // These are the only members used to evaluate operation expressions
+    .op_name = PCMK_ACTION_MONITOR,
+    .op_interval_ms = 10000,
+};
+
+/*!
+ * \internal
+ * \brief Run one test, comparing return value
+ *
+ * \param[in] xml_string    Operation expression XML as string
+ * \param[in] reference_rc  Assert that evaluation result equals this
+ */
+static void
+assert_op_expression(const char *xml_string, int reference_rc)
+{
+    xmlNode *xml = pcmk__xml_parse(xml_string);
+
+    assert_int_equal(pcmk__evaluate_op_expression(xml, &rule_input),
+                     reference_rc);
+    free_xml(xml);
+}
+
+
+/*
+ * Invalid arguments
+ */
+
+#define EXPR_NO_MATCH                                       \
```

Comment:
We don't meaningfully use this. The only use is in a case where `rule_input == NULL`.

The name `EXPR_NO_MATCH` suggests that we care about mismatches in both the name and the interval, but we never actually check either one.

Could just drop this and use a different XML string constant in `null_invalid()` if you want